contracts/strategy/BnbxYieldConverterStrategy.sol

```diff
@@ -0,0 +1,245 @@
+    uint256 private _bnbxToUnstake; // amount of bnbx to withdraw from stader in next batchWithdraw
+    uint256 private _bnbToDistribute; // amount of bnb to distribute to users who unstaked
+
+    event StakeManagerChanged(address stakeManager);
+
+    /// @dev initialize function - Constructor for Upgradable contract, can be only called once during deployment
+    /// @param destination Address of the stakeManager contract
+    /// @param feeRecipient Address of the fee recipient, which receives yield
+    /// @param bnbxToken Address of BNBx token
+    /// @param masterVault Address of the masterVault contract
+    /// @param stakeManager Address of stakeManager contract
+    function initialize(
+        address destination,
+        address feeRecipient,
+        address bnbxToken,
+        address masterVault,
+        address stakeManager
+    ) public initializer {
+        __BaseStrategy_init(destination, feeRecipient);
+
+        _bnbxToken = IERC20Upgradeable(bnbxToken);
+        _stakeManager = IStakeManager(stakeManager);
+        _vault = IMasterVault(masterVault);
+
+        _bnbxToken.approve(stakeManager, type(uint256).max);
+    }
+
+    /**
+     * Modifiers
+     */
+    modifier onlyVault() {
+        require(msg.sender == address(_vault), "!vault");
+        _;
+    }
+
+    /// @dev deposits the given amount of BNB into Stader stakeManager through bnbxRouter
+    function deposit() external payable onlyVault returns (uint256) {
+        uint256 amount = msg.value;
+        require(amount <= address(this).balance, "insufficient balance");
+
+        return _deposit(amount);
+    }
+
+    /// @dev deposits all the available BNB into Stader stakeManager through bnbxRouter
+    function depositAll() external payable onlyVault returns (uint256) {
+        return _deposit(address(this).balance);
+    }
+
+    /// @dev internal function to deposit the given amount of BNB into Stader stakeManager through bnbxRouter
+    /// @param amount amount of BNB to deposit
+    function _deposit(uint256 amount)
+        internal
+        whenDepositNotPaused
+        returns (uint256)
+    {
+        require(canDeposit(amount), "invalid amount");
+
+        _bnbDepositBalance += amount;
+        _bnbxHoldingBalance += _stakeManager.convertBnbToBnbX(amount);
+        _stakeManager.deposit{value: amount}();
+        return amount;
+    }
+
+    /// @dev withdraws the given amount of BNB from Stader's stakeManager
+    /// @param amount amount of BNB to withdraw
+    function withdraw(address recipient, uint256 amount)
+        external
+        onlyVault
+        returns (uint256)
+    {
+        return _withdraw(recipient, amount);
+    }
+
+    /// @dev withdraws everything from Stader's stakeManager
+    function panic() external onlyStrategist returns (uint256) {
+        (, , uint256 debt) = _vault.strategyParams(address(this));
+        return _withdraw(address(_vault), debt);
```

Comment:
How can we give away the power to withdraw any BNB to the strategist when all BNB incoming to this contract from the stakeManager has to be distributed to individual recipients?
Consider the following cases
1. Any BNB deposited immediately gets converted to BNBx.
2. Any BNB that gets transferred gets rejected (as there's no receive fn)

if debt is tracked by the master vault, why then are we tracking bnbDeposited?

Reply:
Is the expectation that this panicked BNB will not be distributed to users later?

Does the master vault have bookkeeping to track this intervention?

Reply:
There is a `receive` fn here.

master has the ability to update debt i.e. `debt = any number`
[updateDebt](https://github.com/helio-money/helio-smart-contracts/blob/5cacfb85b973613c0da06304d56c6484cf969c8e/contracts/masterVault/MasterVault.sol#L212)

Reply:
> Is the expectation that this panicked BNB will not be distributed to users later?

I don't think so. We can check with Helio folks while PR review.

> Does the master vault have bookkeeping to track this intervention?

Couldn't find any bookkeeping for this. Will check with Helios folks during PR review again.
But they do have a waiting pool [contract](https://github.com/helio-money/helio-smart-contracts/blob/add-masterVault/contracts/masterVault/WaitingPool.sol)
which has ability to maually add withdraw entries and pay off users debts

Reply:
pstake and Helio [discussion](https://github.com/helio-money/helio-smart-contracts/pull/9#discussion_r1013989015)

Reply:
As discussed with Dheeraj, 
This is okay as it is not messing with bnbToDistribute or our book-keeping
Whereas it would just create a entry in our book-keeping for masterVault.